src/init/features/project.ts

```diff
@@ -0,0 +1,133 @@
+import * as clc from "cli-color";
+import * as _ from "lodash";
+
+import { FirebaseProject, getProject, listProjects } from "../../firebaseApi";
+import * as logger from "../../logger";
+import { promptOnce, Question } from "../../prompt";
+import * as utils from "../../utils";
+
+const NO_PROJECT = "[don't setup a default project]";
+const NEW_PROJECT = "[create a new project]";
+
+/**
+ * Used in init flows to keep information about the project - basically
+ * a shorter version of FirebaseProject with some additional fields.
+ */
+export interface ProjectInfo {
+  id: string; // maps to projectId
+  label: string;
+  instance: string; // maps to FirebaseProject.resources.realtimeDatabaseInstance
+  location: string; // maps to FirebaseProject.resources.locationId
+}
+
+/**
+ * Get the user's desired project, prompting if necessary.
+ * Returns a ProjectInfo object:
+ *
+ * {
+ *  id: project ID [required]
+ *  label: project display label [optional]
+ *  instance: project database instance [optional]
+ * }
+ */
+async function getProjectInfo(options: any): Promise<ProjectInfo> {
+  // The user passed in a --project flag directly, so no need to
+  // load all projects.
+  if (options.project) {
+    // tslint:disable-next-line:no-shadowed-variable
+    let project: FirebaseProject;
```

Comment:
What is `project` shadowing?

Reply:
Oh, it's _way_ down there. Can this just be moved to a higher level? Adding all the disabling makes me uncomfortable

Reply:
I couldn't find where this was complaining - thanks. Removed all the disable lines.